test/stepDefinitions/create-caseSteps.js

```diff
@@ -179,10 +179,18 @@ defineSupportCode(function ({ Given, When, Then}) {
   });
 
   Then(/^I will be on the '(.*)' page$/, async function (expectedPageHeader) {
-    let pageHeader = await caseWizardPage.getPageHeader();
-    expect(pageHeader).to.equal(expectedPageHeader);
+    await IAmOnPageWithHeader(expectedPageHeader)
+  })
+
+  Then(/^the '(.*)' page should be displayed as a result of the fixed list show condition$/, async function(expectedPageHeader) {
```

Comment:
looking at the code, this should be: ```/^the '(.*)' page should be displayed``` only. The part ``` as a result of the fixed list show condition``` limits the reusability of this step. As a result we will have to introduce many duplicated versions in the future

Reply:
sure- but there comes a point where you need to strike the balance between giving details of what is happening in a step in the context of the whole scenario over re-usability, abstracting the main code into a function means you are not having to write duplicated code while being able to express more details in the bdd